pkg/gateway/multiparts/tracker.go

```diff
@@ -38,82 +72,70 @@ var (
 	ErrInvalidMetadataSrcFormat = errors.New("invalid metadata source format")
 )
 
-func NewTracker(adb db.Database) Tracker {
+func NewTracker(ms kv.StoreMessage) Tracker {
 	return &tracker{
-		db: adb,
+		store: ms,
+	}
+}
+
+func multipartFromProto(pb *MultipartUploadData) *MultipartUpload {
```

Comment:
This is great - nothing actionable - just a thought that maybe in the next iteration we can also support as part of the model's method. Similar to how the current DB can query for 'Scan' method to convert the structure to JSON. We can do the same to get from/to proto format.